astropy/table/tests/test_table.py

```diff
@@ -2392,3 +2392,15 @@ class MyTable(table.Table):
         assert type(t['c']) is MyMaskedColumn
         assert type(t['d']) is MySubColumn
         assert type(t['e']) is MySubMaskedColumn  # sub-class not downgraded
+
+
+def test_iterrows():
+    t = table.Table([(2, 3, 2, 1), (8, 7, 6, 5)], names=('a', 'b'))
```

Comment:
Use values that are easily verified by a human to be right, e.g. `[(1, 2, 3), (4, 5, 6)]`.